src/main/java/uk/gov/hmcts/darts/audio/service/impl/MediaRequestServiceImpl.java

```diff
@@ -500,4 +513,59 @@ public List<AdminMediaSearchResponseItem> adminMediaSearch(Integer transformedMe
             && mediaEntity.getEnd().isAfter(mediaRequest.getStartTime())).toList();
         return AdminMediaSearchResponseMapper.createResponseItemList(filteredMediaList, hearing);
     }
+
+    @Override
+    public MediaHideResponse adminHideOrShowMediaById(Integer mediaId, MediaHideRequest mediaHideRequest) {
+        MediaHideResponse response;
+
+        IdRequest<MediaHideRequest> request = new IdRequest<>(mediaHideRequest, mediaId);
+        mediaHideOrShowValidator.validate(request);
+
+        Optional<MediaEntity> mediaEntityOptional
+            = mediaRepository.findById(mediaId);
+        if (mediaEntityOptional.isPresent()) {
+            MediaEntity mediaEntity = mediaEntityOptional.get();
+
+            mediaEntity.setHidden(mediaHideRequest.getIsHidden());
+            mediaEntity = mediaRepository.saveAndFlush(mediaEntity);
+
+            if (request.getPayload().getIsHidden()) {
+                Optional<ObjectHiddenReasonEntity> objectHiddenReasonEntity;
+                objectHiddenReasonEntity = objectHiddenReasonRepository.findById(mediaHideRequest.getAdminAction().getReasonId());
+
+                if (objectHiddenReasonEntity.isEmpty()) {
+                    throw new DartsApiException(AudioApiError
+                                                    .MEDIA_HIDE_ACTION_REASON_NOT_FOUND);
+                }
+
+                // on hiding add the relevant hide record
+                ObjectAdminActionEntity objectAdminActionEntity = new ObjectAdminActionEntity();
+                objectAdminActionEntity.setObjectHiddenReason(objectHiddenReasonEntity.get());
+                objectAdminActionEntity.setTicketReference(mediaHideRequest.getAdminAction().getTicketReference());
+                objectAdminActionEntity.setComments(mediaHideRequest.getAdminAction().getComments());
+                objectAdminActionEntity.setMedia(mediaEntity);
+                objectAdminActionEntity.setHiddenBy(userIdentity.getUserAccount());
+                objectAdminActionEntity.setHiddenDateTime(OffsetDateTime.now());
+                objectAdminActionEntity.setMarkedForManualDeletion(false);
+                objectAdminActionEntity.setMarkedForManualDelBy(userIdentity.getUserAccount());
+                objectAdminActionEntity.setMarkedForManualDelDateTime(OffsetDateTime.now());
+
+                objectAdminActionEntity = objectAdminActionRepository.saveAndFlush(objectAdminActionEntity);
```

Comment:
Is it possible to get here if the `media` is already hidden (and the request tries to hide it again)?

Reply:
Just seen your int test, so I don't think it's possible - ignore.